core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -137,6 +137,12 @@
      */
     TableStatistics getTableStatistics(Session session, TableHandle tableHandle, Constraint constraint);
 
+    /**
+     * Returns {@code true} when {@link #getTableStatistics} returns more useful (e.g. more accurate) result
+     * for {@code oldTableHandle} than for {@code newTableHandle}.
+     */
+    boolean degradesTableStatisticsAccuracy(Session session, List<TableHandle> oldTableHandles, TableHandle newTableHandle);
```

Comment:
Why do we pass multiple "old" table handles here?

Reply:
for Join pushdown case.

Reply:
We discussed this offline a bit with @findepi.
 
Another option to attack this is to get rid `degradesTableStatisticsAccuracy` method at all.
Istead we can add a flag `shouldEngineProvideStatsForTableHandle` (naming to change) in `*Result` classes used as return types of `applyAggregation`, `applyJoin` ...

IMO flow would be easier to understand then.